File: 2900.longest-unequal-adjacent-groups-subsequence-i.kt
// @leet start
class Solution {
    fun getLongestSubsequence(words: Array<String>, groups: IntArray): List<String> {
      var r = mutableListOf<String>()
      var j = 0
      r.add(words[j])
      for (i in groups.indices) {
        if (groups[i] != groups[j]) {
          j = i
          r.add(words[j])
        }
      }
      return r
    }
}
// @leet end

File: 2900.longest-unequal-adjacent-groups-subsequence-i.ts
// @leet start
function getLongestSubsequence(words: string[], groups: number[]): string[] {
  let res: string[] = [words[0]];
  let j = 0;
  for (let i = 1; i < words.length; i++) {
    if (groups[i] !== groups[j]) {
      j = i;
      res.push(words[j]);
    }
  }
  return res;
};
